### PR TITLE
Use Util Nuget package, delete x86+StrongName configs and modernize projects

### PR DESCRIPTION
**Pull request by sisiplac**
### TODOs:
- [x]  Wait until Util release is added to nuget.org
- [x] Delete x86/AnyCPU solution config
- [x] Delete StrongName config (build strong name by default if possible)
- [x] Unite NetStandard and non-NetStandard projects
- [x] Reference `Metrilus.Util` in C++/CLI projects differently using the following path:
`<HintPath>..\..\bin\$(Configuration)\net472\Metrilus.Util.dll</HintPath>`
- [x] Use `Metrilus.Util` Nuget reference in all projects
- [x] Delete 3rd party license files (same as for Metrilus.Util)
- [x] Remove `MetrilusReferencesVersions.props` (no more needed) and all `lib*` directories
- [x] Remove separate Strong-Name configuration and just build Signed where possible.
- [x] Upgrade all single target framework projects to .NET 4.7.2
- [x] Update Jenkinsfile due to new folder structure of multi target framework builds.

### Useful links:
* Good description on how to properly use solution-wide version control of one reference:
https://www.strathweb.com/2018/07/solution-wide-nuget-package-version-handling-with-msbuild-15/
https://docs.microsoft.com/de-de/visualstudio/msbuild/customize-your-build?view=vs-2019
* Unfortunately, the usage of `PackageReference` is not possible for C++/Cli projects:
https://developercommunity.visualstudio.com/idea/351636/use-packagereference-in-vcxproj.html
Since the `packages.config` mechanism is crap, the C++/CLI projects should consume the unpacked DLLs in the `MetriCam2` output folder as long as `PackageReference` is not supported.
* It is not so easy to make custom Nuget feeds conditional (only if network share really exists in our case), but here is a good description:  
https://github.com/NuGet/Home/issues/3972
https://github.com/NuGet/Home/issues/6045

### CODE REVIEW

**Comment by f00f** [comment]
C++/CLI: Why not consume from the `packages` folder?

**Comment by f00f** [comment]
I think splitting this into several PRs - e.g. delete x86, delete strongname, upgrade .Net version, use nuget, and probably some more - would have been the faster overall process. Here are my reasons:

- Each PR would be smaller. Smaller PRs get reviewed quicker, because the reviewer doesn't have to reserve a day for it.
- Most of those PRs could have been reviewed already instead of all waiting for the Util nuget package.
- Other features which should way for some changes from here could start already, e.g. creating MetriCam nuget packages should be done after those big changes to the project files, but could be started before MetriCam uses the Util nuget package.

**Comment by sisiplac** [comment]
> C++/CLI: Why not consume from the `packages` folder?

This is motivated by a couple of reasons:
1) I guess our Metrilus.Util package would require some modification to support C++/CLI projects. When trying to use the current package I get: `Could not install package 'Metrilus.Util 16.1.0-master.521'. You are trying to install this package into a project that targets 'native,Version=v0.0', but the package does not contain any assembly references or content files that are compatible with that framework. For more information, contact the package author`
2) The `packages.config` mechanism would hardcode the Metrilus.Util version number in the project files of the C++/CLI projects - so we would need to patch every C++/CLI project file.
3) Using the "unpacked" Nuget package of the MetriCam2 project is save (its a project dependency) and is better than the current `lib`-solution, which requires manual copying of the libraries. So all in all this is an improvement.
4) I would not waste our time in finding hard-to-maintain workarounds for 1 and 2. I think it would make more sense to migrate still needed C++/CLI projects to C# - then we could also compile to multiple target platforms including `netstandard` (with the benefit of being able to support Linux+Android for e.g. Orbbec).

**Comment by sisiplac** [comment]
> I think splitting this into several PRs - e.g. delete x86, delete strongname, upgrade .Net version, use nuget, and probably some more - would have been the faster overall process.

I did it just the way i would have liked it as a reviewer. Reviewing each file up to four times would be quite annoying (use nuget reference, remove x86, remove strong name, switch to new .csproj format). I think it makes e.g. no sense to review content changes on `MetriCam2.NetStandard.csproj` or `RealSense2.NetStandard.csproj`, and then see that both projects are deleted in the final step, when all projects are migrated to the new csproj-format.  The new csproj file format is so compact that the review should not take too long.
If you like/can imagine to be the reviewer, I can give you a good review-introduction tomorrow and support you where I can ;-)

**Comment by f00f** [comment]
> > C++/CLI: Why not consume from the `packages` folder?
> 
> This is motivated by a couple of reasons:
> 
> [ ... ]

Okay. I hadn't realized that the new mechanism doesn't create the `packages` folder anymore.

**Comment by sisiplac** [comment]
@f00f The PR can be reviewed now! I will create open issues for versioning and Nuget package creation.

**Comment by f00f** [comment]
Die Solution hat nur noch `x64`.  
Das SimpleViewer Projekt hat nur noch `Any CPU`, welches aber als `x86` konfiguriert ist. Urgs.  
Gleiches beim MinimalSample.

**Comment by f00f** [comment]
Is K4A x64 only?

**Comment by sisiplac** [comment]
> Is K4A x64 only?

It supports both x64 and x86, but no AnyCPU.

**Comment by sisiplac** [comment]
> Die Solution hat nur noch `x64`.
> Das SimpleViewer Projekt hat nur noch `Any CPU`, welches aber als `x86` konfiguriert ist. Urgs.
> Gleiches beim MinimalSample.

Die beiden Programme laufen aber mit der x64 Kinect4Azure - wieso geht das dann?? Die Warning müssten wir wegbringen, indem wir explizit für `x64` bauen - soll ich das mal eben machen und comitten?

**Comment by f00f** [comment]
> > Die Solution hat nur noch `x64`.
> > Das SimpleViewer Projekt hat nur noch `Any CPU`, welches aber als `x86` konfiguriert ist. Urgs.
> > Gleiches beim MinimalSample.
> 
> Die beiden Programme laufen aber mit der x64 Kinect4Azure - wieso geht das dann?? Die Warning müssten wir wegbringen, indem wir explizit für `x64` bauen - soll ich das mal eben machen und comitten?

Ja. Bitte änder die Config auf x64 und das Dropdown der Pltaform target auch. Danke

**Comment by f00f** [comment]
> > Is K4A x64 only?
> 
> It supports both x64 and x86, but no AnyCPU.

Hmpf. Das sin ja Granaten.

File: BetaCameras/Kinect2/Kinect2.csproj [hunk 1]
```diff
@@ -1,117 +1,28 @@
...
-    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
-    <DebugType>full</DebugType>
-    <PlatformTarget>AnyCPU</PlatformTarget>
-    <ErrorReport>prompt</ErrorReport>
-    <CodeAnalysisRuleSet>MinimumRecommendedRules.ruleset</CodeAnalysisRuleSet>
-    <SignAssembly>true</SignAssembly>
-    <DelaySign>false</DelaySign>
-    <AssemblyOriginatorKeyFile>Z:\protection\StrongNameKey\StrongName.snk</AssemblyOriginatorKeyFile>
+    <Description>Wrapper for Kinect2 camera</Description>
+    <Product>MetriCam2: Kinect2 wrapper</Product>
```
Comment (f00f) [inline BetaCameras/Kinect2/Kinect2.csproj]: We usually include the vendor.
```suggestion
    <Product>MetriCam2: Microsoft Kinect2 wrapper</Product>
```

File: BetaCameras/Kinect4Azure/Kinect4Azure.cs [hunk 2]
```diff
@@ -11,6 +10,10 @@
 using MetriCam2.Exceptions;
 using Microsoft.Azure.Kinect.Sensor;
 using System.Buffers;
+#if NETSTANDARD2_0
```
Comment (f00f) [inline BetaCameras/Kinect4Azure/Kinect4Azure.cs]:
```suggestion
#if !NETSTANDARD2_0
```
(and remove the `#else` line below)
(L195 uses `!`, too)

File: BetaCameras/Kinect4Azure/Kinect4Azure.csproj [hunk 3]
```diff
@@ -1,105 +1,27 @@
...
-    <DebugType>full</DebugType>
-    <PlatformTarget>x64</PlatformTarget>
-    <ErrorReport>prompt</ErrorReport>
-    <CodeAnalysisRuleSet>MinimumRecommendedRules.ruleset</CodeAnalysisRuleSet>
-  </PropertyGroup>
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Release|x64'">
-    <OutputPath>..\..\bin\x64\Release\</OutputPath>
-    <DefineConstants>TRACE</DefineConstants>
+    <Description>Wrapper for Microsoft Kinect 4 Azure (K4A) cameras</Description>
+    <Product>MetriCam2: Kinect4Azure wrapper</Product>
```
Comment (f00f) [inline BetaCameras/Kinect4Azure/Kinect4Azure.csproj]:
```suggestion
    <Product>MetriCam2: Microsoft Azure Kinect wrapper</Product>
```

File: BetaCameras/Kinect4Azure/Kinect4Azure.csproj [hunk 4]
```diff
@@ -1,105 +1,27 @@
...
-    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
-    <DebugType>full</DebugType>
-    <PlatformTarget>x64</PlatformTarget>
-    <ErrorReport>prompt</ErrorReport>
-    <CodeAnalysisRuleSet>MinimumRecommendedRules.ruleset</CodeAnalysisRuleSet>
-  </PropertyGroup>
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Release|x64'">
-    <OutputPath>..\..\bin\x64\Release\</OutputPath>
-    <DefineConstants>TRACE</DefineConstants>
+    <Description>Wrapper for Microsoft Kinect 4 Azure (K4A) cameras</Description>
```
Comment (f00f) [inline BetaCameras/Kinect4Azure/Kinect4Azure.csproj]:
```suggestion
    <Description>Wrapper for Microsoft Azure Kinect (K4A) cameras</Description>
```

File: BetaCameras/RealSense/RealSense.csproj [hunk 5]
```diff
@@ -1,120 +1,30 @@
...
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Release|x64'">
-    <OutputPath>..\..\bin\x64\Release\</OutputPath>
-    <DefineConstants>TRACE</DefineConstants>
-    <Optimize>true</Optimize>
-    <DebugType>pdbonly</DebugType>
-    <PlatformTarget>x64</PlatformTarget>
-    <ErrorReport>prompt</ErrorReport>
-    <CodeAnalysisRuleSet>MinimumRecommendedRules.ruleset</CodeAnalysisRuleSet>
+    <Description>Wrapper for RealSense cameras</Description>
+    <Product>MetriCam2: RealSense wrapper</Product>
```
Comment (f00f) [inline BetaCameras/RealSense/RealSense.csproj]:
```suggestion
    <Product>MetriCam2: Intel RealSense wrapper</Product>
```

File: BetaCameras/RealSense/RealSense.csproj [hunk 6]
```diff
@@ -1,120 +1,30 @@
...
-  </PropertyGroup>
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Release|x64'">
-    <OutputPath>..\..\bin\x64\Release\</OutputPath>
-    <DefineConstants>TRACE</DefineConstants>
-    <Optimize>true</Optimize>
-    <DebugType>pdbonly</DebugType>
-    <PlatformTarget>x64</PlatformTarget>
-    <ErrorReport>prompt</ErrorReport>
-    <CodeAnalysisRuleSet>MinimumRecommendedRules.ruleset</CodeAnalysisRuleSet>
+    <Description>Wrapper for RealSense cameras</Description>
```
Comment (f00f) [inline BetaCameras/RealSense/RealSense.csproj]:
```suggestion
    <Description>Wrapper for Intel RealSense cameras</Description>
```

File: BetaCameras/RealSense2/RealSense2.csproj [hunk 7]
```diff
@@ -1,103 +1,33 @@
...
-    <TargetFrameworkProfile />
-    <BaseIntermediateOutputPath>obj_netFramework\</BaseIntermediateOutputPath>
-  </PropertyGroup>
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug|AnyCPU'">
-    <DebugSymbols>true</DebugSymbols>
-    <OutputPath>..\..\bin\Debug\</OutputPath>
-    <DefineConstants>DEBUG;TRACE</DefineConstants>
+    <AssemblyName>MetriCam2.Cameras.RealSense2</AssemblyName>     
+    <Description>Wrapper for Intel libRealSense2 cameras</Description>
+    <Product>MetriCam2: libRealSense2 wrapper</Product>
```
Comment (f00f) [inline BetaCameras/RealSense2/RealSense2.csproj]:
```suggestion
    <Product>MetriCam2: Intel libRealSense2 wrapper</Product>
```

File: BetaCameras/Kinect4Azure/Kinect4Azure.cs (hunk 2 above)
Comment (sisiplac) [reply BetaCameras/Kinect4Azure/Kinect4Azure.cs]: Is it only possible to modify one line when providing suggestions?

File: BetaCameras/BaslerACE/BaslerACE.csproj [hunk 8]
```diff
@@ -1,97 +1,30 @@
...
+    <Product>MetriCam 2: BaslerACE wrapper"</Product>
+    <TargetFrameworks>net472;net45</TargetFrameworks>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <Platforms>x64</Platforms>
   </PropertyGroup>
   <ItemGroup>
-    <Reference Include="Basler.Pylon, Version=1.0.0.0, Culture=neutral, PublicKeyToken=e389355f398382ab, processorArchitecture=AMD64">
-      <SpecificVersion>False</SpecificVersion>
+    <Reference Include="Basler.Pylon">
       <HintPath>Z:\external-libraries\Basler\acA1300\pylon 5.0.11\Basler.Pylon\x64\Basler.Pylon.dll</HintPath>
```
[the remark this answers is not in the document]
Comment (sisiplac) [reply BetaCameras/BaslerACE/BaslerACE.csproj]: I have moved your proposal to #210

**Comment by sisiplac** [comment]
I guess you will need to set some "reviewed"-checkbox, so that I am able to merge.

**Pull request merged**